components/backup-stream/src/router.rs

```diff
@@ -964,69 +991,167 @@ impl StreamTaskInfo {
                 info!("remove template file"; "err" => ?e);
             }
         }
+        for (_, v, _) in self.flushing_meta_files.write().await.drain(..) {
+            let data_file = v.lock().await;
+            debug!("removing meta data file"; "size" => %data_file.file_size, "name" => %data_file.local_path.display());
+            self.total_size
+                .fetch_sub(data_file.file_size, Ordering::SeqCst);
+            if let Err(e) = data_file.remove_temp_file().await {
+                // if remove template failed, just skip it.
+                info!("remove template file"; "err" => ?e);
+            }
+        }
     }
 
-    async fn flush_log_file_to(
+    async fn merge_and_flush_log_files_to(
         storage: Arc<dyn ExternalStorage>,
-        file: &Mutex<DataFile>,
+        files: &[(TempFileKey, Mutex<DataFile>, DataFileInfo)],
+        metadata: &mut MetadataInfo,
+        is_meta: bool,
     ) -> Result<()> {
-        let data_file = file.lock().await;
+        // save locks for longer lifetime
+        let mut data_files = Vec::new();
```

Comment:
the variable `data_files` is unused?

Reply:
Until the flush is done, the files are used to upload. So I want to keep files locked.

Reply:
Remove the `Slot`, everything looks good!